test/modules/core/lifecycle/component-state.spec.js

```diff
@@ -9,5 +18,103 @@ test('ComponentState#imports', t => {
 test('ComponentState#synchronous async props', t => {
   const state = new ComponentState();
   t.ok(state, 'ComponentState construction ok');
+
+  t.equals(state.hasAsyncProp('data'), false, 'ComponentState.hasAsyncProp returned correct value');
+  state.setAsyncProps({data: []});
+  state.setAsyncProps({data: []});
+  t.equals(state.hasAsyncProp('data'), true, 'ComponentState.hasAsyncProp returned correct value');
+  t.deepEquals(
+    state.getAsyncProp('data'),
+    [],
+    'ComponentState.getAsyncProp returned correct value'
+  );
   t.end();
 });
+
+test('ComponentState#asynchronous async props', t => {
```

Comment:
Add test cases for synchronized data props?

Reply:
> Add test cases for synchronized data props?

Good idea. This class is focused on handling the async props only so would require some extra plumbing here to just check that others are ignored.

I think the best approach is to add some `Component`, rather than `ComponentState` level tests for that, will consider for a next PR.